packages/transport/src/modules/relationshipTemplates/RelationshipTemplateController.ts

```diff
@@ -251,9 +329,28 @@ export class RelationshipTemplateController extends TransportController {
         const relationshipTemplate = RelationshipTemplate.from({
             id: id,
             secretKey: secretKey,
-            isOwn: false
+            isOwn: false,
+            passwordProtection
         });
-        await this.updateCacheOfTemplate(relationshipTemplate);
+
+        try {
+            await this.updateCacheOfTemplate(relationshipTemplate);
+        } catch (e) {
+            if (passwordProtection && e instanceof RequestError && e.status === 404) {
+                throw new RequestError(
+                    e.method,
+                    e.path,
+                    e.platformParameters,
+                    e.code,
+                    "RelationshipTemplate not found. Make sure the ID exists, the record is not expired and the entered password is correct.",
+                    e.requestId,
+                    e.status,
+                    e.time,
+                    e.object
+                );
```

Comment:
I don't like this approach. Why do you throw a RequestError instead of a record not found by the runtime? Just altering one thing from the backbone and make it look like the backbone outputs another error message is very weird looking and very hard to understand.